### DISCUSSION TITLE
Hosting: "Uncaught SyntaxError: Unexpected token <"

### DISCUSSION THREAD

**Discussion by kevinXmichael** [quasarframework/quasar#8751] [CLI - PWA mode] [answered]
My quasar webapp is in PWA mode and I deploy with firebase using cache-control to always re-validate and an age of 0. But still when I deploy a new version and visit a dynamic site of my web app, the first load gives me the error:
`Uncaught SyntaxError: Unexpected token <` as it does no find the "main script" because it is always named randomly.

How can I fix this? Any ideas? I found a lot of "solutions" on GitHub but none of them work.

**Comment by metalsadman** [comment]
You will need some mechanism that detects that the source has been updated, otherwise you'll be left doing manually thru your browser.

**Comment by metalsadman** [comment]
See this thread https://forum.quasar-framework.org/topic/2560/solved-pwa-force-refresh-when-new-version-released/55.

**Comment by kevinXmichael** [comment, marked as answer]
The solution is to add the following in the `register-service-worker.js`:
```
updatefound (registration) {
    try {
      registration.update()
    } catch (err) {
      console.err('SW update failed:', err)
    }
  }
```

**Reply by saifobeidat** [reply]
did it work with you? and do you reload the site if there's a new content?

**Reply by kevinXmichael** [reply]
Yes it works, the new version I pushed needed a "hard reload" again to have the new logic. But with that update all new versions seem to load the update properly.

**Reply by ontwikkelfabriek** [reply]
@kmschott would you be so kind to mark the post as answered.
